pkg/providers/instanceprofile/suite_test.go

```diff
@@ -103,14 +108,167 @@ var _ = Describe("InstanceProfileProvider", func() {
 	DescribeTable(
 		"should support IAM roles",
 		func(roleWithPath, role string) {
-			const profileName = "test-profile"
+			const profileName = "profile-A"
 			nodeClass.Spec.Role = roleWithPath
-			Expect(awsEnv.InstanceProfileProvider.Create(ctx, profileName, role, nil)).To(Succeed())
+			Expect(awsEnv.InstanceProfileProvider.Create(ctx, profileName, role, nil, string(nodeClass.UID))).To(Succeed())
 			Expect(profileName).ToNot(BeNil())
 			Expect(awsEnv.IAMAPI.InstanceProfiles[profileName].Roles).To(HaveLen(1))
 			Expect(aws.ToString(awsEnv.IAMAPI.InstanceProfiles[profileName].Roles[0].RoleName)).To(Equal(role))
 		},
 		Entry("with custom paths", fmt.Sprintf("CustomPath/%s", nodeRole), nodeRole),
 		Entry("without custom paths", nodeRole, nodeRole),
 	)
+
+	It("should list all instance profiles for a NodeClass", func() {
```

Comment:
I would simplify these test cases such that we're not testing the functionality of `Create` and `List` in the same test. E.g. I'd just add the instance profiles to the mock manually and verify that List returns the correct result. We have additional tests to verify `Creates` functionality.